repair/repair.cc

```diff
@@ -2090,20 +2121,45 @@ static future<> do_decommission_removenode_with_repair(seastar::sharded<database
                 ranges.swap(ranges_for_removenode);
             }
             auto nr_ranges_synced = ranges.size();
-            sync_data_using_repair(db, ms, keyspace_name, std::move(ranges), std::move(range_sources), reason).get();
+            sync_data_using_repair(db, ms, keyspace_name, std::move(ranges), std::move(range_sources), reason, ops->ops_uuid).get();
             rlogger.info("{}: finished with keyspace={}, leaving_node={}, nr_ranges={}, nr_ranges_synced={}, nr_ranges_skipped={}",
                 op, keyspace_name, leaving_node, nr_ranges_total, nr_ranges_synced, nr_ranges_skipped);
         }
         rlogger.info("{}: finished with keyspaces={}, leaving_node={}", op, keyspaces, leaving_node);
     });
 }
 
+
+static thread_local std::unordered_map<utils::UUID, lw_shared_ptr<node_ops>> _node_ops;
+
```

Comment:
Is there no better place for this? It's better to put things inside sharded<> instances rather than globals that then have to be disentangled by @xemul .

Reply:
I can put it inside the repair_tracker.

Reply:
Removed the global in new version.